### PR TITLE
[JENKINS-43668] - Remove Trilead references

### PR DESCRIPTION
**Pull request by mc1arke**
[JENKINS-43610] Move to latest version of ssh-cli-auth to remove Trilead references from SSHD module

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
   <properties>
     <jenkins.version>1.580.3</jenkins.version>
-    <java.level>6</java.level>
+    <java.level>7</java.level>
```
Comment (oleg-nenashev) [inline pom.xml]: Also needs bumping the core dependency then. I would propose to pick 1.625.x as a first version with Java 7

File: pom.xml [hunk 2]
```diff
@@ -62,20 +62,20 @@
     <dependency>
       <groupId>org.jenkins-ci.modules</groupId>
       <artifactId>ssh-cli-auth</artifactId>
-      <version>1.1</version>
+      <version>1.4</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Also requires 1.625.3

Comment (mc1arke) [inline pom.xml]: Updated.

I've just gone with the latest release version, but 1.2 would also work. Would there be any reason to stick with a lower version?

File: src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java [hunk 3]
```diff
@@ -34,36 +29,13 @@ public boolean authenticate(String username, PublicKey key, ServerSession sessio
             return false;
         }
 
-        // TODO: use sshKey.has() when we can depend on 1.446 or later
-        if (!sshKey.isAuthorizedKey(getPublicKeySignature(key))) {
-            LOGGER.fine("Key signature didn't match for the user: "+username+" : "+getPublicKeySignature(key));
+        if (!sshKey.has(key)) {
+            LOGGER.fine("Key signature didn't match for the user: "+username);
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java]: Would it be possible to continue printing some diagnostics message from `key`?

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: The only potential issue is that backporting of this change (e.g. SSHD Module 1.12 or above) will also require bumping the ssh-cli-auth module. I do not see any issue with that in this case. The modules have to evolve anyway

File: src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java (hunk 3 above)
Comment (mc1arke) [inline src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java]: Sorry, I missed this comment. We can probably print out the whole public key, but the signature generation relies on Trilead which is what we're wanting to remove, so I'm not keen on still trying to generate a signature for logging purposes.

Comment (mc1arke) [inline src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java]: Signature now retained for the log message, as requested

**Comment by oleg-nenashev** [comment]
Maybe it makes sense to convert JENKINS-43610 to EPIC and to add tasks for module. But it is up to you, the current approach is fine for me if you prefer it

**Comment by mc1arke** [comment]
Agreed on the Epic. JENKINS-43668 now covers this task

**Comment by oleg-nenashev** [comment]
Should be merged without squash from what I see.
@mc1arke Are you fine if I leave the commit link to the second PR? Or would you mind changing the commit message?

**Comment by mc1arke** [comment]
I'll amend my commits later today, and will probably squash the last 3 commits into one, so I have one for fixing the local build, and one for using the new signature generation mechanism; unless anyone objects?

File: pom.xml [hunk 4]
```diff
@@ -51,20 +51,20 @@
     <dependency>
       <groupId>org.jenkins-ci.modules</groupId>
       <artifactId>ssh-cli-auth</artifactId>
-      <version>1.2</version>
+      <version>1.4</version>
```
Comment (jglick) [inline pom.xml]: To pick up https://github.com/jenkinsci/ssh-cli-auth-module/pull/2 IIUC?

Comment (oleg-nenashev) [inline pom.xml]: yes

**Pull request merged**